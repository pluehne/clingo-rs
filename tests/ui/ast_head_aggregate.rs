use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term = ast::Term::from(sym);
    let mut guard = ast::AggregateGuard::gt(term);
    let elements = vec![];
    let hagg = ast::HeadAggregate::new(
        ast::AggregateFunction::Count,
        &elements,
        Some(&guard),
        Some(&guard),
    );
    guard =  ast::AggregateGuard::gt(term);
    drop(elements);
    let _end = (guard, hagg);
}
