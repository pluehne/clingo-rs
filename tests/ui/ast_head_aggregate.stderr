error[E0506]: cannot assign to `guard` because it is borrowed
  --> $DIR/ast_head_aggregate.rs:14:5
   |
11 |         Some(&guard),
   |              ------ borrow of `guard` occurs here
...
14 |     guard =  ast::AggregateGuard::gt(term);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `guard` occurs here
15 |     drop(elements);
16 |     let _end = (guard, hagg);
   |                        ---- borrow later used here

error[E0505]: cannot move out of `elements` because it is borrowed
  --> $DIR/ast_head_aggregate.rs:15:10
   |
10 |         &elements,
   |         --------- borrow of `elements` occurs here
...
15 |     drop(elements);
   |          ^^^^^^^^ move out of `elements` occurs here
16 |     let _end = (guard, hagg);
   |                        ---- borrow later used here
